include/socket.h

```diff
@@ -0,0 +1,134 @@
+/***************************************************************************//**
```

Comment:
Does this file make [tcp_socket.h](https://github.com/analogdevicesinc/no-OS/blob/master/include/tcp_socket.h) redundant?

Reply:
Yes, tcp_socket.h is removed in one commit